File: syris_core/transport/models/filters.py
from typing import Optional, Set, List
from pydantic import BaseModel, Field

from .enums import EventKind, Level

class StreamSubscription(BaseModel):
    name: str = Field(min_length=1, max_length=64)
    kinds: Optional[Set[EventKind]] = None
    levels: Optional[Set[Level]] = None

    include_payload: bool = True
    sample_rate: float = Field(default=1.0, ge=0.0, le=1.0)

class TransportFilters(BaseModel):
    request_id: Optional[str] = Field(default=None, max_length=128)
    entity_id: Optional[str] = Field(default=None, max_length=256)
    entity_prefix: Optional[str] = Field(default=None, max_length=256)

    kinds: Optional[Set[EventKind]] = None
    levels: Optional[Set[Level]] = None

class SubscribeOptions(BaseModel):
    include_recent: bool = True
    recent_limit: int = Field(default=200, ge=0, le=10_000)

File: archive/syris_ui/types/filters.ts
import type { EventKind, Level } from "./enums";

export type StreamSubscription = {
  name: string; // "logs" | "events" | etc.
  kinds?: EventKind[];
  levels?: Level[];

  include_payload?: boolean;
  include_payload_raw?: boolean;
  sample_rate?: number; // 0..1
};

export type TransportFilters = {
  request_id?: string | null;
  entity_id?: string | null;
  entity_prefix?: string | null;

  kinds?: EventKind[];
  levels?: Level[];

  // optional future filters
  integration_id?: string | null;
  tool_name?: string | null;
};

export type SubscribeOptions = {
  include_recent?: boolean;
  recent_limit?: number; // 0..10000
};
